osm/src/main/groovy/org/orbisgis/osm/Loader.groovy

```diff
@@ -0,0 +1,115 @@
+package org.orbisgis.osm
+
+import groovy.transform.BaseScript
+import org.orbisgis.datamanager.JdbcDataSource
+import org.orbisgis.processmanagerapi.IProcess
+import groovy.json.JsonSlurper
+
+import java.security.InvalidParameterException
+
+
+@BaseScript OSMHelper osmHelper
+
+
+/**
+ * This process extracts OSM data as an XML file using the Overpass API
+ * @param overpassQuery The overpass api to be executed
+ *
+ * @author Erwan Bocher CNRS LAB-STICC
+ * @author Elisabeth Lesaux UBS LAB-STICC
+ */
+static IProcess extract() {
+    return processFactory.create(
+            "Extract the OSM data using the overpass api and save the result in an XML file",
+            [
+             overpassQuery :String
+            ],
+            [outputFilePath : String],
+            { overpassQuery  ->
+                        logger.info('Extract the OSM data')
+                        def tmpOSMFile = File.createTempFile("extract_osm", ".osm")
+                        def osmFilePath =  tmpOSMFile.absolutePath
+                        if(executeOverPassQuery(overpassQuery, tmpOSMFile)){
+                            logger.info("The OSM file has been downloaded at ${osmFilePath}.")
+                        }
+                        else{
+                            logger.error("Cannot extract the OSM data for the query $overpassQuery")
+                            return
+                        }
+                [outputFilePath : osmFilePath]
+            }
+    )
+}
+
+
+
+
+/**
+ * This process is used to load an OSM file in a database.
+ *
+ * @param datasource A connection to a database
+ * @param osmTablesPrefix A prefix to identify the 10 OSM tables
+ * @param omsFilePath The path where the OSM file is
+ * @return datasource The connection to the database
+ * @author Erwan Bocher CNRS LAB-STICC
+ * @author Elisabeth Lesaux UBS LAB-STICC
+ */
+static IProcess load() {
+    return processFactory.create(
+            "Load an OSM file to the current database",
+            [datasource: JdbcDataSource,
+             osmTablesPrefix: String,
+             osmFilePath :String
+            ],
+            [datasource : JdbcDataSource],
+            { datasource, osmTablesPrefix, osmFilePath  ->
+                if (datasource != null) {
+                    logger.info('Load the OSM file in the database')
+                    File osmFile = new File(osmFilePath)
+                    if (osmFile.exists()) {
+                        datasource.load(osmFile.absolutePath, osmTablesPrefix, true)
```

Comment:
you can assert that `datasource.load()` return a not null value in case of success